python/mxnet/image/image.py

```diff
@@ -132,6 +134,9 @@ def imdecode(buf, *args, **kwargs):
     <NDArray 224x224x3 @cpu(0)>
     """
     if not isinstance(buf, nd.NDArray):
+        if sys.version_info[0] == 3:
+            if isinstance(buf, str):
+                buf = bytes(buf, "ascii")
```

Comment:
mx.base.py_str is 'utf8' encoding,I'm not sure it can work.

Reply:
how about using buffer?

Reply:
This case really shouldn't happen, because in this case the API is underspecified.

String is always tied to an encoding, and if the string comes from the user, it could have been decoded with any encoding/charset, so the 'ascii' assumption can easily be wrong.

Reply:
OK